### ISSUE TITLE
None result should be None (fetchrow and fetch)

### ISSUE THREAD

**Issue by achimnol** [CanopyTax/asyncpgsa#16]
```python
query = ...  # some SA select query
row = await conn.fetchrow(query)  # conn is SAConnection
```

When the underlying asyncpg connection returns None as the "empty" result, `row` should become None as well.
Currently, `row` is *not* None but `row.row` is None.
This would confuse the users who expect the API semantics to be same to asyncpg.

The same applies to `fetch` and `execute` which returns a `RecordGenerator` because it does not check None when instantiating `Record` objects.

Q: What's the purpose of having seem-to-be-redundant `Record` and `RecordGenerator` classes?
They look like a no-op proxy to asyncpg's return objects.

**Comment by nhumrich** [comment]
> Q: What's the purpose of having seem-to-be-redundant Record and RecordGenerator classes?
> They look like a no-op proxy to asyncpg's return objects.

Probably sounds kind of stupid, but they convert a dictionary reference `row['column'] to a dot reference `row.column`. This might be an optional thing in the future if people like the standard dictionary lookup. 

In this specific usecase where you are fetching a single row, the response being None makes sense. I'll see what I can do.

I apologize for the terribly late response, my notification settings were set up incorrectly.

**Comment by nhumrich** [comment]
Upon further looking, `__bool__` is implemented to check if the row is empty
So you should be able to use `if row` in your code, and it will always check for emptiness. Dont need to check `if row.row is None`

**Comment by achimnol** [comment]
Ah, I missed __bool__ method as I did empty check using None comparison in asyncpg. It would be nice to have some docs for such differences or make the return value compatible.

**Comment by nhumrich** [comment]
Ya, thats valid. I'll leave this open until I do something about it.

**Comment by rbcarson** [comment]
When using `row = await conn.fetchrow(q)`, I can confirm that `if row` works as expected. However, `__bool__` isn't implemented for `RecordGenerator`, so for something like `rows = await conn.fetch(q)`,  `if rows` will always be true. I've found myself doing `if rows.data` instead - is this alternative actually reliable?

**Comment by nhumrich** [comment]
@rbcarson Good catch. This is fixed in 0.8.2. RecordGenerator will now be falsey on an empty list.

**Comment by achimnol** [comment]
I think this could be closed now, as I'm good with the new `__bool__` methods.
